src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -123,4 +127,39 @@ object HttpGoogleServicesDAO {
     storage.objects().get(bucketName, objectKey).executeMediaAsInputStream
   }
 
+  // create a GCS signed url as per https://cloud.google.com/storage/docs/access-control/create-signed-urls-program
+  def getSignedUrl(bucketName: String, objectKey: String) = {
+
+    // generate the string-to-be-signed
+    val verb = "GET"
+    val md5 = ""
+    val contentType = ""
+    val expireSeconds = (System.currentTimeMillis() / 1000) + 120 // expires 2 minutes (120 seconds) from now
+    val objectPath = s"/$bucketName/$objectKey"
+
+    val signableString = s"$verb\n$md5\n$contentType\n$expireSeconds\n$objectPath"
+
+    // use GoogleCredential.Builder to parse the private key from the pem file
+    val builder = new GoogleCredential.Builder()
+      .setServiceAccountPrivateKeyFromPemFile(new java.io.File(rawlsPemFile))
+    val privateKey = builder.getServiceAccountPrivateKey
+
+    // sign the string
+    val signature = java.security.Signature.getInstance("SHA256withRSA")
+    signature.initSign(privateKey)
+    signature.update(signableString.getBytes("UTF-8"))
+    val signedBytes = signature.sign()
```

Comment:
Is there no way to sign as the user? The obvious problem is security which you have jumped through some hoops deal with. The other problem is that this won't work if the rawls service account does not have access while the user does (protected TCGA data may fall into this category, I am not sure).